src/crimson/os/seastore/segment_cleaner.h

```diff
@@ -440,6 +516,140 @@ class SegmentCleaner : public JournalSegmentProvider {
       config.max_journal_segments);
     return ret;
   }
+
+  // GC status helpers
+  paddr_t gc_current_pos = P_ADDR_NULL;
+
+  /**
+   * do_gc
+   *
+   * Performs bytes worth of gc work on t.
+   */
+  using do_gc_ertr = SegmentManager::read_ertr;
+  using do_gc_ret = do_gc_ertr::future<>;
+  do_gc_ret do_gc(
+    Transaction &t,
+    size_t bytes);
+
+  size_t get_bytes_used_current_segment() const {
+    assert(journal_head != journal_seq_t());
+    return journal_head.offset.offset;
+  }
+
+  size_t get_bytes_available_current_segment() const {
+    return config.segment_size - get_bytes_used_current_segment();
+  }
+
+  /**
+   * get_bytes_gcd_current_segment
+   *
+   * Returns the number of bytes from the current gc segment that
+   * have been scanned.
+   */
+  size_t get_bytes_gcd_current_segment() const {
```

Comment:
nit, to me, this name is a little bit difficult to digest. as "gcd" is also short for "greatest common divisor". with this in mind, "bytes", "gcd", and "current segments" are all nouns.

how about renaming it to `get_gc_offset()` or `get_gc_progress()`? as i don't see the needs to differentiate the "current" gc segment from the next or previous gc segment in this context.

Reply:
I really want the _bytes_ in there -- quite a few of these are ratios.  I'll go with get_bytes_scanned_current_segment.